File: app/src/main/java/com/cyrillrx/uremote/explorer/LocalExplorer.java
package com.cyrillrx.uremote.explorer;

import com.cyrillrx.notifier.Toaster;
import com.cyrillrx.uremote.R;
import com.cyrillrx.uremote.request.FileInfoFactory;
import com.cyrillrx.uremote.request.protobuf.RemoteCommand;

/**
 * @author Cyril Leroux
 *         Created 14/05/2017.
 */
public class LocalExplorer extends Explorer {

    public LocalExplorer(Callback callback) { super(callback); }

    @Override
    public void navigateTo(String dirPath) {
        super.navigateTo(dirPath);

        if (dirPath == null) {
            Toaster.toast(R.string.msg_no_path_defined);
            return;
        }

        currentFileInfo = FileInfoFactory.createFromLocalPath(dirPath, null, true);
        callback.onDirectoryLoaded(currentFileInfo);
    }
}

File: app/src/main/java/com/cyrillrx/uremote/explorer/LocalExplorer.kt
package com.cyrillrx.uremote.explorer

import com.cyrillrx.uremote.request.FileInfoFactory

/**
 * @author Cyril Leroux
 *         Created 14/05/2017.
 */
class LocalExplorer(callback: Explorer.Callback) : Explorer(callback) {

    override fun navigateTo(dirPath: String) {
        super.navigateTo(dirPath)

        currentFileInfo = FileInfoFactory.createFromLocalPath(dirPath, null, true)
        callback.onDirectoryLoaded(currentFileInfo!!)
    }
}
